### ISSUE TITLE
[Magento 2.3.3] Uncaught TypeError: Return value of Magento\CatalogUrlRewrite\Model\Storage\DynamicStorage::getCategoryUrlSuffix() must be of the type string, null returned

### ISSUE THREAD

**Issue by hostep** [magento/magento2#24903]
Will update issue when I find some more time, I'm just quickly creating this so I don't forget, not tested yet on vanilla install and steps to reproduce might not be accurate.

### Preconditions (*)
1. Magento 2.3.3 (is most likely also reproducible on 2.3-develop since the problematic code is in there)
2. PHP 7.2.20

### Steps to reproduce (*)
1. Setup Magento 2.3.3
2. Configure:
    - Catalog > Catalog > Search Engine Optimization > Category URL Suffix => (empty)
    - Catalog > Catalog > Search Engine Optimization > Generate "category/product" URL Rewrites => No
3. Visit some category page on the frontend

### Expected result (*)
1. Category page loads

### Actual result (*)
1. Category page crashes with error:
```

Fatal error: Uncaught TypeError: Return value of Magento\CatalogUrlRewrite\Model\Storage\DynamicStorage::getCategoryUrlSuffix() must be of the type string, null returned in vendor/magento/module-catalog-url-rewrite/Model/Storage/DynamicStorage.php:150 Stack trace:
#0 vendor/magento/module-catalog-url-rewrite/Model/Storage/DynamicStorage.php(235): Magento\CatalogUrlRewrite\Model\Storage\DynamicStorage->getCategoryUrlSuffix(1)
#1 vendor/magento/module-catalog-url-rewrite/Model/Storage/DynamicStorage.php(133): Magento\CatalogUrlRewrite\Model\Storage\DynamicStorage->findProductRewritesByFilter(Array)
#2 vendor/magento/module-url-rewrite/Model/Storage/AbstractStorage.php(44): Magento\CatalogUrlRewrite\Model\Storage\DynamicStorage->doFindAllByData(Array)
#3 generated/code/Magento/CatalogUrlRewrite/Model/Storage/DynamicStorage/Interceptor.php(37): Mage in vendor/magento/module-catalog-url-rewrite/Model/Storage/DynamicStorage.php on line 150
```

### Problem & Solution 

Issue was introduced in [MC-4244](https://github.com/magento/magento2/commit/211dd25c73b#diff-eb1d8c6c0059c0e71a5cbdc544a4558f)

ScopeConfig's `getValue` method returns `null` for this setting and the code guarantees the `getCategoryUrlSuffix` method should return a string, so php doesn't like this.

Solution:

```php
    private function getCategoryUrlSuffix($storeId = null): string
    {
        return $this->config->getValue(
            CategoryUrlPathGenerator::XML_PATH_CATEGORY_URL_SUFFIX,
            ScopeInterface::SCOPE_STORE,
            $storeId
        ) ?? '';
    }
```

Will create PR later on if nobody already has.

**Comment by BrentRobert** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @BrentRobert. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @BrentRobert, here is your [Magento instance](https://i-24903-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-24903-2-3-develop.instances.magento-community.engineering/admin](https://i-24903-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by BrentRobert** [comment]
@hostep I can not reproduce with the steps provided, however I see this is indeed an issue as a null value is possible in the config. I created a PR with the provided fix.

**Comment by hostep** [comment]
Thanks @BrentRobert!

Unfortunately it's really hard to find exact steps to reproduce, it's because of a bunch of data in our `url_rewrite` table which somehow causes it I think, but trying to reproduce this on a vanilla installation is really hard. I'm going to give it up for now.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-21676` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by saark-akash** [comment]
@magento please provide the magento instance

**Comment by saark-akash** [comment]
@magento please provide the magento instance to test

**Comment by saark-akash** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @akashwebkul. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @akashwebkul, here is your [Magento instance](https://i-24903-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-24903-2-3-develop.instances.magento-community.engineering/admin](https://i-24903-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by saark-akash** [comment]
@magento Not able to reproduce from the mentioned steps.

**Comment by saark-akash** [comment]
@magento give me 2.3.3 instance

**Comment by magento-engcom-team** [comment]
Hi @akashwebkul. Thank you for your request. I'm working on Magento 2.3.3 instance for you

**Comment by magento-engcom-team** [comment]
Hi @akashwebkul, here is your [Magento instance](https://i-24903-2-3-3.instances.magento-community.engineering).
Admin access: [https://i-24903-2-3-3.instances.magento-community.engineering/admin](https://i-24903-2-3-3.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by hostep** [comment]
@akashwebkul: just FYI, a pull request is already in the works: https://github.com/magento/magento2/pull/24907

**Comment by hostep** [comment]
Since the bot seems to no longer auto-close issues today, let me do the honors myself :)

----

Hi @hostep. Thank you for your report.
The issue has been fixed in #24907 by @BrentRobert in 2.3-develop branch
Related commit(s):

- 04305166519d90a01e9233c17eb6b4f2cd49b153

The fix will be available with the upcoming 2.3.4 release.
